tests/specs/mcd/verification.k

```diff
@@ -138,6 +160,13 @@ module LEMMAS-MCD-HASKELL [symbolic, kore]
     rule #range(M:Memory [ K <- V ] [ K' := VS':ByteArray ], START, WIDTH) => #range(M [ K := (V : VS')        ], START, WIDTH)  requires K' +Int #sizeByteArray(VS') ==Int START +Int WIDTH andBool START <=Int K andBool 0 <Int WIDTH andBool 0 <=Int START
                                                                                                                                   andBool K +Int 1                    ==Int K'                                                                                [concrete(K, K', V, VS'), simplification]
 
+    rule { (M:Map [K <- V1]) #Equals (M [K <- V2]) } => { V1 #Equals V2 }
+         [simplification]
+        
+    rule { M:Map [K <- V1] #Equals M } => 
+         { true #Equals K in_keys(M) andBool V1 ==K #lookup (M, K) }
+         [simplification]
```

Comment:
```suggestion
    rule { M:Map [K <- V1] #Equals M } => #Top requires K in_keys(M) andBool V1 ==K #lookup (M, K)
         [simplification]
```

Does this work? It keeps the original structure of the problem if it can't simplify, which has smaller term size.

Reply:
I guess it won't work if you want the RHS to simplify to `#Bottom` instead.

Reply:
Yes, I think I would like to have both options.